### PR TITLE
deposit form: add sharing of drafts

### PR DESCRIPTION
**Pull request by anikachurilova**
* closes https://github.com/inveniosoftware/invenio-app-rdm/issues/2667
* closes https://github.com/zenodo/zenodo/issues/151

Updated dropdowns:
Screenshot 2024-05-08 at 17 24 49
Screenshot 2024-05-08 at 17 25 12
Screenshot 2024-05-08 at 17 25 24

Share button on the deposit form:
enabled:
Screenshot 2024-05-08 at 17 26 20
disabled:
Screenshot 2024-05-08 at 17 26 47

### CODE REVIEW

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html [hunk 1]
```diff
@@ -41,6 +41,11 @@
            value='{{ files | tojson }}'>
   {%- endif %}
 
+  {%- if config %}
+    <input type="hidden" name="deposits-groups-enabled"
+           value='{{ config.USERS_RESOURCES_GROUPS_ENABLED | tojson }}'>
+  {%- endif %}
+
```
Comment (ntarocco) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/deposit.html:48]: Config is always there, and I would also change the name:
```suggestion
  <input type="hidden" name="config-groups-enabled"
           value='{{ config.USERS_RESOURCES_GROUPS_ENABLED | tojson }}'>

```

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ import { FundingField } from "@js/invenio_vocabularies";
 import { Card, Container, Grid, Ref, Sticky } from "semantic-ui-react";
 import PropTypes from "prop-types";
 import Overridable from "react-overridable";
+import { ShareDraftButton } from "../landing_page/ShareOptions/ShareDraftButton";
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:46]: We should maybe move the button away from `landing_page/`, to a more generic/global components module

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js [hunk 3]
```diff
@@ -614,6 +621,16 @@ export class RDMDepositForm extends Component {
                           <Grid.Column width={16} className="pt-10">
                             <PublishButton fluid />
                           </Grid.Column>
+
+                          <Grid.Column width={16} className="pt-0">
+                            {(record.is_draft === null || permissions.can_manage) && (
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:626]: Can it be `null`? Shouldn't it be true/false?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/index.js [hunk 4]
```diff
@@ -22,6 +22,7 @@ ReactDOM.render(
       config={getInputFromDOM("deposits-config")}
       permissions={getInputFromDOM("deposits-record-permissions")}
       filesLocked={getInputFromDOM("deposits-record-locked-files")}
+      groupsEnabled={getInputFromDOM("deposits-groups-enabled")}
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/index.js:25]:
```suggestion
      groupsEnabled={getInputFromDOM("config-groups-enabled")}
```

**Review by ntarocco — APPROVED** [review]
Clean PR, good job! 👏🏼 If you have time, you might want to clean up some of the other linting issues reported in the PR ;)

**Comment by kpsherva** [comment]
1. The view goes blank if I try to share metadata only unsaved draft
2. while granting access:
Screenshot 2024-05-14 at 14 51 44

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js (hunk 3 above)
Comment (anikachurilova) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js:626]: `is_draft` is null after just clicking on the "New upload" and rendering an empty deposit form

File: invenio_app_rdm/records_ui/views/deposits.py [hunk 5]
```diff
@@ -361,6 +361,8 @@ def new_record():
     defaults = current_app.config.get("APP_RDM_DEPOSIT_FORM_DEFAULTS") or {}
     for key, value in defaults.items():
         set_default_value(record, value, key)
+
+    record["parent"]["access"] = {"grants": [], "links": []}
```
Comment (kpsherva) [inline invenio_app_rdm/records_ui/views/deposits.py:365]: shouldn't this part be handled by dump_empty in 354 line?

**Pull request merged**